### PR TITLE
Deflake TestAgentForward

### LINKED ISSUE #5333: Flaky tests: lib/srv/regular, time-sensitive assertion
Time-sensitive assertion failure.

<details><summary><code>Logs</code></summary>

```
----------------------------------------------------------------------
FAIL: sshserver_test.go:371: SrvSuite.TestAgentForward

sshserver_test.go:456:
    c.Fatalf("expected socket to be closed, still could dial after 150 ms")
... Error: expected socket to be closed, still could dial after 150 ms

OOPS: 23 passed, 1 skipped, 1 FAILED
--- FAIL: TestRegular (17.51s)
FAIL
coverage: 61.2% of statements
FAIL	github.com/gravitational/teleport/lib/srv/regular	17.661s
```
</details>

### PR DESCRIPTION
**Pull request by zmb3**
This was another case of the t.TempDir() being cleaned up while the
audit logger is still writing to the directory, which happens when
tests don't properly clean up after themselves. Ensure that any
services we spin up closed via cleanup actions.

Updates #9492
Fixes #5333

### CODE REVIEW

File: lib/srv/regular/sshserver_test.go [hunk 1]
```diff
@@ -678,14 +682,12 @@ func TestAgentForward(t *testing.T) {
...
-		}
-		time.Sleep(50 * time.Millisecond)
-	}
-	require.FailNow(t, "expected socket to be closed, still could dial after 150 ms")
+	require.Eventually(t, func() bool {
+		_, err := net.Dial("unix", socketPath)
+		return err != nil
+	},
+		150*time.Millisecond, 50*time.Millisecond,
+		"expected socket to be closed, still coud dial")
```
Comment (Joerger) [inline lib/srv/regular/sshserver_test.go:690]:
```suggestion
		"expected socket to be closed, still could dial")
```

**Review by Joerger — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Comment by zmb3** [comment]
@Joerger and @r0mant I think I've fixed this for good now. See latest commit.

**Review by Joerger — APPROVED** [review]

**Comment by zmb3** [comment]
Well, I've improved things and changed the error.

It used to reproduce after 2-3 minutes with the tempdir cleanup error.

Now it takes 10+ minutes but occasionally fails with a different error:
```
--- FAIL: TestAgentForward (3.04s)
    sshserver_test.go:625: 
                Error Trace:    sshserver_test.go:625
                Error:          Received unexpected error:
                                EOF
                Test:           TestAgentForward
FAIL
FAIL    github.com/gravitational/teleport/lib/srv/regular       3.711s
FAIL
```

**Comment by zmb3** [comment]
Interesting thing is the godocs say:

```
// RequestAgentForwarding sets up agent forwarding for the session.
// ForwardToAgent or ForwardToRemote should be called to route
// the authentication requests.
```

Yet nowhere in the tests do we call `ForwardToAgent` or `ForwardToRemote`.
I"m not familiar enough with this code to know if that's a problem.

**Comment by zmb3** [comment]
This fixes the flakiness, but I noticed a separate issue which I've filed here: https://github.com/gravitational/teleport/issues/13335

**Review by xacrimon — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**